core/client/src/client.rs

```diff
@@ -1179,6 +1179,48 @@ impl<B, E, Block, RA> Client<B, E, Block, RA> where
 		Ok(None)
 	}
 
+	fn get_descendants(&self, target: Block::Hash, maybe_skip: Option<Block::Hash>) -> error::Result<Vec<Block::Hash>> {
+		let mut children = self.backend.blockchain().children(target)?;
+		if let Some(skip) = maybe_skip {
+			children.retain(|&c| c != skip);
+		}
+		let mut descendants = Vec::new();
+		for child in children {
+			descendants.push(child);
+			let d = self.get_descendants(child, None)?;
```

Comment:
won't this recurse all the way to the head of all descendant chains? could be expensive.